pkg/cmd/provisioning/constants.go

```diff
@@ -0,0 +1,33 @@
+package provisioning
+
+const (
+	// PrivateKeyFile is the name of the private key file created by "ccoctl create key-pair" command
+	PrivateKeyFile = "serviceaccount-signer.private"
+	// PublicKeyFile is the name of the public key file created by "ccoctl create key-pair" command
+	PublicKeyFile = "serviceaccount-signer.public"
+	// DiscoveryDocumentURI is a URI for the OpenID configuration discovery document
+	DiscoveryDocumentURI = ".well-known/openid-configuration"
+	// KeysURI is a URI for public key that enables client to validate a JSON Web Token issued by the Identity Provider
+	KeysURI = "keys.json"
+	// ManifestsDirName is the name of the directory to save installer manifests created by ccoctl
+	ManifestsDirName = "manifests"
+	// TLSDirName is the name of the directory to save bound service account signing key created by ccoctl
+	TLSDirName = "tls"
+
+	// AWS
+
+	// CcoctlAWSResourceTagKeyPrefix is the prefix of the tag key applied to the AWS resources created/shared by ccoctl
+	CcoctlAWSResourceTagKeyPrefix = "openshift.io/cloud-credential-operator"
+	// OwnedCcoctlAWSResourceTagValue is the value of the tag applied to the AWS resources created by ccoctl
+	OwnedCcoctlAWSResourceTagValue = "owned"
+	// NameTagKey is the key of the "Name" tag applied to the AWS resources created by ccoctl
+	NameTagKey = "Name"
+	// Generated identity provider files
+	OidcBucketFilename          = "01-oidc-bucket.json"
+	OidcConfigurationFilename   = "02-openid-configuration"
+	OidcKeysFilename            = "03-keys.json"
+	IAMIdentityProviderFilename = "04-iam-identity-provider.json"
+	// Generated role files
+	RoleFilenameFormat       = "05-%d-%s-role.json"
+	RolePolicyFilenameFormat = "06-%d-%s-policy.json"
```

Comment:
nit: no reason to keep aws specific constants in a global package. keep them private and close to the usage in aws implementation.

Reply:
These are the constants which are shared among different files in cloud provider packages. There are two options here
1) to create one constants file for entire provisioning package
2) to create separate constants file for each cloud provider

I prefer option (1) as it is easy to manage

Reply:
if these are truly global constants, then the var name `CcoctlAWSResourceTagKeyPrefix` shouldn't say `AWS` in the var name. Same for `OwnedCcoctlAWSResourceTagValue`.
And the comments for those two vars and `NameTagKey` shouldn't mention AWS.

Reply:
`CcoctlAWSResourceTagKeyPrefix` and `OwnedCcoctlAWSResourceTagValue` are constants specific to aws , but they are shared between different files in provisioning/aws package. I think I should move them somewhere inside the package.